auth/oidc_provider.go

```diff
@@ -132,12 +133,15 @@ func (pc *OidcProviderConfiguration) parseURI(s string) (*url.URL, error) {
 		return nil, nil
 	}
 	u, err := url.Parse(s)
-	if err == nil {
-		if u.Host == "" {
-			return nil, fmt.Errorf("Host required in URI [%s]:", s)
-		} else if u.Scheme != "http" && u.Scheme != "https" {
-			return nil, fmt.Errorf("Invalid URI scheme [%s]:", s)
-		}
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "Error parsing URI: [%s]", s)
```

Comment:
Previously this code was (incorrectly) not returning the error in this case.  Have we done any verification on whether making this fix breaks anything in our oidc support?

Reply:
I didn't do any verification, and I don't see any unit tests that cover this code (http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-coverage/16/artifact/cover_sg_merged.html#file3).

Reply:
I pushed a commit that adds a test that ensures that the common providers won't cause any regressions by returning URLs that don't parse, which could cause regressions due to the error checking change.

Reply:
That looks like a good approach - thanks.